src/components/editor/document-bar/permissions/permission-modal.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from 'react'
+import { Alert, Modal } from 'react-bootstrap'
+import { Trans, useTranslation } from 'react-i18next'
+import { getUserById } from '../../../../api/users'
+import { CommonModal } from '../../../common/modals/common-modal'
+import { ShowIf } from '../../../common/show-if/show-if'
+import { UserAvatar, UserAvatarProps } from '../../../common/user-avatar/user-avatar'
+import { GroupMode, PermissionGroupEntry } from './permission-group-entry'
+import { PermissionList } from './permission-list'
+
+export interface PermissionsModalProps {
+  show: boolean,
+  onChangeShow: (newShow: boolean) => void
+}
+
+export interface Principal {
+  id: string
+  name: string
+  photo: string
+  canEdit: boolean
+}
+
+interface NotePermissions {
+  owner: string
+  sharedTo: {
+    username: string
+    canEdit: boolean
+  }[],
+  sharedToGroup: {
+    id: string
+    canEdit: boolean
+  }[]
+}
+
+const permissionsApiResponse: NotePermissions = {
+  owner: 'dermolly',
+  sharedTo: [{
+    username: 'emcrx',
+    canEdit: true
+  }, {
+    username: 'mrdrogdrog',
+    canEdit: false
+  }],
+  sharedToGroup: [{
+    id: '1',
+    canEdit: true
+  }, {
+    id: '2',
+    canEdit: false
+  }]
+}
+
+export const PermissionModal: React.FC<PermissionsModalProps> = ({ show, onChangeShow }) => {
+  useTranslation()
+  const [error, setError] = useState(false)
+  const [userList, setUserList] = useState<Principal[]>([])
+  const [owner, setOwner] = useState<UserAvatarProps>()
+  const [allPermissions, setAllPermissions] = useState(GroupMode.NONE)
+  const [loggedPermissions, setLoggedPermissions] = useState(GroupMode.NONE)
```

Comment:
Maybe it's just me, but I don't like these names. In my opinion they should indicate that they belong to these special groups "allUsers"/"everyone" and "loggedInUsers".

Reply:
done

Reply:
done